pkg/controllermanager/controller/seed/seed_lifecycle_reconcile.go

```diff
@@ -147,14 +142,14 @@ func (c *livecycleReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 
 	seedLogger.Debugf("Gardenlet didn't send a heartbeat for at least %s - setting the shoot conditions/constraints to 'unknown' for all shoots on this seed", c.config.Controllers.Seed.ShootMonitorPeriod.Duration)
 
-	shootList, err := c.shootLister.List(labels.Everything())
-	if err != nil {
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := c.gardenClient.Client().List(ctx, shootList); err != nil {
```

Comment:
Uff, this one is quite heavy for server and client. Can we read from the cache here? Maybe also using a field selector for `spec.seedName`?